### DISCUSSION TITLE
how to convert json object to a simple list

### DISCUSSION THREAD

**Discussion by stemuser027** [nlohmann/json#2355] [Q&A] [answered]
I am new to this area. I am trying to use json library https://github.com/nlohmann/json to parse out data in the JSON feed , and construct a simple list that the user go through with keyboard 

How do I do this? 

This is what I know so far based on your documentation
The segment below reads the contents of the JSON file into JSON object "jsonObject".

// read a JSON file
std::ifstream i("file.json");
json jsonObject;
i >> jsonObject;

How do I convert jsonObject into a simple list?
Please help.

**Comment by nlohmann** [comment]
How does the JSON file look like? What do you mean with “simple list”?

**Comment by stemuser027** [comment, marked as answer]
Thank you so much for your response.

This is an example of the json file

http://statsapi.mlb.com/api/v1/schedule?hydrate=game(content(editorial(recap))),decisions&date=2018-06-10&sportId=1

I just want to be able to parse the file, display the contents and use the up,down,left and right arrows on the key board to scroll through the formatted list.

I truly appreciate you taking the time to do this.

**Reply by nlohmann** [reply]
Assuming you downloaded the JSON file to `stats.json`, here is some code to show how it's parsed and traversed:

```cpp
#include <fstream>
#include <iostream>
#include "json.hpp"

using json = nlohmann::json;

int main()
{
    std::ifstream stats_file("stats.json");
    json stats = json::parse(stats_file);
    
    for (const auto& date : stats["dates"])
    {
        std::cout << date["date"] << std::endl;
        
        for (const auto& game : date["games"])
        {
            std::string home_team = game["teams"]["home"]["team"]["name"];
            std::string away_team = game["teams"]["away"]["team"]["name"];
            std::cout << home_team << " vs. " << away_team << std::endl;
        }
    }
}
```

Output:

```
"2018-06-10"
Boston Red Sox vs. Chicago White Sox
Toronto Blue Jays vs. Baltimore Orioles
Detroit Tigers vs. Cleveland Indians
Cincinnati Reds vs. St. Louis Cardinals
Tampa Bay Rays vs. Seattle Mariners
Miami Marlins vs. San Diego Padres
Philadelphia Phillies vs. Milwaukee Brewers
Minnesota Twins vs. Los Angeles Angels
Chicago Cubs vs. Pittsburgh Pirates
Texas Rangers vs. Houston Astros
Colorado Rockies vs. Arizona Diamondbacks
Oakland Athletics vs. Kansas City Royals
Washington Nationals vs. San Francisco Giants
Los Angeles Dodgers vs. Atlanta Braves
New York Mets vs. New York Yankees
```

**Comment by stemuser027** [comment]
Thank you so much for the information. I need your help again. I am getting an error.

I copied the contents from the url feed into a file called "stats.json", and put it in my project directory.

It builds fine but when I try to execute, I get the following error:

terminate called after throwing an instance of 'nlohmann::detail::parse_error'
  what():  [json.exception.parse_error.101] parse error at line 1, column 1: syntax error while parsing value - unexpect
ed end of input; expected '[', '{', or a literal

Process finished with exit code 3

**Reply by nlohmann** [reply]
It seems you did not successfully open the file. Can you please double check the file is opened in the right directory, and that you can actually read from it?

**Comment by stemuser027** [comment]
Thank you so much for your help. 

Is there anyway I can read and parse the contents of this website directly? I know I can read from a file, but what about from a website like the example below?
I will truly appreciate your help.

This is an example of the json file

http://statsapi.mlb.com/api/v1/schedule?hydrate=game(content(editorial(recap))),decisions&date=2018-06-10&sportId=1

**Comment by stemuser027** [comment]
I have a question for you.

For this discussion I will use a file called help.cpp

When I use CLION IDE, and use
#include <json.hpp>

it works fine. I am able to build help.cpp and run my program siuccessfully.

If I try to build the same help.cpp file on ubuntu, it fails and complains about <json.hpp>.
When I change it to 
#include <nlohmann/json.hpp>, I am able to build help.cpp sucessfully, but when I try to run the program,

I get the follow error:

Cannot open the files for reading
terminate called after throwing an instance of 'nlohmann::detail::parse_error'
  what():  [json.exception.parse_error.101] parse error at line 1, column 1: syntax error while parsing value - unexpected end of input; expected '[', '{', or a literal
Aborted (core dumped)

On Ubuntu I installed the following json:
sudo apt update
sudo apt install nlohmann3-json-dev

I hope you can help me. I have spent all day working on it. I have been up for over 18 hours and I am concerned I will cause more harm than good at this point.

Please help.

**Comment by stemuser027** [comment]
OK...

I definitely need to go to bed. I believe I just found the issue.

This is the line of code I have to read the file:

std::string input_file = R"(C:\Users\Josep\CLionProjects\untitled5\stats.json)";

ubuntu does not recognize the path so I copied the file into my project directory and it worked.

Thank you

**Reply by nlohmann** [reply]
Is there anything left to do in this issue?
